### PR TITLE
[path_provider] Allow getDownloadsPath for Windows

### PR DESCRIPTION
**Pull request by technolion**
## Description

The path_provider_plugin for Windows is still in the FDE repository and works well but it misses the `getDownloadsPath `call on Windows which is already available on macOS.
I created a PR that adds the missing call in flutter-desktop-embedding: https://github.com/google/flutter-desktop-embedding/pull/765
PR 765 has to land first.

_This_ PR is about allowing the call on Windows. Code changes here are trivial, which is why I incremented the version number only on the minor.

## Related Issues

flutter/flutter#41715

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by technolion** [comment]
@stuartmorgan  could I please ask for this to get merged? It's just a very minor change that allows using `getDownloadsPath` on Windows. So others can make use of this until #41715 gets fixed.

File: packages/path_provider/path_provider_platform_interface/lib/src/method_channel_path_provider.dart [hunk 1]
```diff
@@ -78,8 +78,9 @@ class MethodChannelPathProvider extends PathProviderPlatform {
   }
 
   Future<String> getDownloadsPath() {
-    if (!_platform.isMacOS) {
-      throw UnsupportedError('Functionality only available on macOS');
+    if (!_platform.isMacOS && !_platform.isWindows) {
```
Comment (stuartmorgan) [inline packages/path_provider/path_provider_platform_interface/lib/src/method_channel_path_provider.dart:81]: This should not be opt-in; it undermines one of the goals of federated plugins. It also doesn't address the fact that Linux is incorrectly blocked here, and fixes for this more general problem shouldn't be ad-hoc.

I don't know why this plugin does pre-screening by platform at this layer in the first place, so someone familiar with the history of this pattern in path_provider will need to review this. If maintainers of the plugin want to keep a check, it should be an opt-out (e.g., return an error for iOS and Android), not opt-in, but my preference would be to follow the more normal plugin pattern of simply allowing the implementation to handle—or not handle—the message.

Comment (technolion) [reply packages/path_provider/path_provider_platform_interface/lib/src/method_channel_path_provider.dart]: I removed the pre-screening by platform as suggested by @stuartmorgan .

**Comment by technolion** [comment]
Googlebot complained:

> We found a Contributor License Agreement for you (the sender of this pull request), but were unable to find agreements for all the commit author(s) or Co-authors.

I simply rebased my branch with the commits from master that had been added after I created this PR initially. Can someone please explain?

**Comment by technolion** [comment]
Is it necessary to rebase a branch to master at all when providing a PR?

**Comment by technolion** [comment]
I am closing this pull request and create a new one.

**Comment by gaaclarke** [comment]
@technolion fyi you can always use `git push -f <remote>` to reset a github PR to any hash you want if you want to keep the conversation of a PR but reset the code.

**Comment by technolion** [comment]
The new PR can be found here: https://github.com/flutter/plugins/pull/3017

**Pull request closed without merging**